ethereum/rlp/src/main/java/org/hyperledger/besu/ethereum/rlp/AbstractRLPInput.java

```diff
@@ -317,6 +317,12 @@ public int readIntScalar() {
     return res;
   }
 
+  @Override
+  public long readUnsignedIntScalar() {
+    checkScalar("unsigned int scalar", 4);
```

Comment:
So you are enforcing that there are no leading zeros now! That means that we are disconnecting older Besu nodes that do not have this PR :-)
I guess it is the right thing to do, it is just funny that we will disconnect older besu nodes ...
What we could do is being lenient with the decoding (allowing the leading zeros), but do the right thing with the encoding, at least for a short time ...

Reply:
Good point, anyway we are not disconnecting other nodes in case of an error, but just skipping this message, so should be safe

Reply:
indeed you are right, and we disconnect other Besus, so I am implementing a backward compatibility decoding

Reply:
made `ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/encoding/TransactionAnnouncementDecoder.java` lenient when reading size